### ISSUE TITLE
es6jsx: Quotes in "single" mode fails for JSX nodes.

### ISSUE THREAD

**Issue by pselden** [eslint/eslint#1477]
This is related to the test case I added here: https://github.com/eslint/eslint/pull/1476

One of my annoyances with the current "jsxhint" package is that it runs the jsxtransform first and then lints the resulting file. Unfortunately JSX transforms the JSX nodes into functions that use double quotes, so any rule for using single quotes fails.

```javascript
<div>Hello world</div>
```

Becomes:

```javascript
React.createElement("div", null, "Hello world")
```
I was hoping that ESLint would not have the same problem but it appears the first pass of it also fails this check. When "quotes": [1, "single"] is on, it reports a warning.
